src/include/migraphx/check_shapes.hpp

```diff
@@ -351,17 +340,13 @@ struct check_shapes
     {
         if(begin == end)
             return false;
-        assert(begin != nullptr);
-        assert(end != nullptr);
         return std::any_of(begin, end, p);
     }
 
     const shape* get(long i) const
```

Comment:
This needs to be update to use the iterator type.

Reply:
No compatible check for both std::iterator and a pointer

Reply:
What check are you talking about?

Reply:
Also, I am not sure how this would compile, because `begin+i` would not be a pointer to shape if `begin` is an iterator.

Reply:
I think we need to differentiate between `std::iterator` and the concept of an iterator. `begin + i` will only work for a random_access_iterator. For that case it will return another `std::iterator` object.